infra/testing/server/index.js

```diff
@@ -31,6 +31,13 @@ function initApp() {
   // Exposed the `.body` property on requests for application/json.
   app.use(bodyParser.json());
 
+  app.use((req, _, next) => {
```

Comment:
One of the big changes in this PR not related to Playwright itself is refactoring the server code under `infra/testing/server/` to make it easier to run multiple tests in parallel. This means doing things like supporting unique URL prefixes that each test could set so that its test state would not interfere with the test state from another test running at the same time.